src/main/java/org/xbill/DNS/TcpKeepaliveOption.java

```diff
@@ -0,0 +1,100 @@
+package org.xbill.DNS;
+
+import java.io.IOException;
+
+/**
+ * TCP Keepalive EDNS0 Option, as defined in https://tools.ietf.org/html/rfc7828
+ *
+ * @see OPTRecord
+ * @author Klaus Malorny
+ */
+public class TcpKeepaliveOption extends EDNSOption {
+
+  /** flag whether the timeout has been provided */
+  private boolean hasTimeout;
+
+  /** the timeout in 100ms units */
+  private int timeout;
```

Comment:
Please use `java.time.Duration` and only encode to 100ms when needed (i.e. in `optionToWire`).